server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1253,8 +1253,8 @@ public VolumeVO resizeVolume(ResizeVolumeCmd cmd) throws ResourceAllocationExcep
 
                 validateIops(newMinIops, newMaxIops, volume.getPoolType());
             } else {
-                newMinIops = newDiskOffering.getMinIops();
-                newMaxIops = newDiskOffering.getMaxIops();
+                newMinIops = newDiskOffering.getMinIops() != null ? newDiskOffering.getMinIops() : newDiskOffering.getIopsReadRate();
+                newMaxIops = newDiskOffering.getMaxIops() != null ? newDiskOffering.getMaxIops() : newDiskOffering.getIopsWriteRate();
```

Comment:
We should extract this to a method and add a javadoc to it explaining why we are doing it like this.